### PR TITLE
fix for https://github.com/codenotary/immudb/issues/679

### PR DESCRIPTION
**Pull request by ellisab**

### CODE REVIEW

**Comment by jeroiraz** [comment]
thanks @ellisab!, I think we can take the opportunity to improve the readability as well.

databasel stands for database name length, it's actually a buffer which is used to finally get the encoded length as a numeric value in variable databaselUint64. Probably both variables may be renamed.

A validation to ensure the length is greater than zero may be convenient as well as checking that remaining content has enough bytes as specified by that length.

What do you think?

File: pkg/client/token_service.go [hunk 1]
```diff
@@ -93,14 +93,17 @@ func (ts *tokenService) parseContent() (string, string, error) {
...
-	databaselUint64 := binary.BigEndian.Uint64(databasel)
-	databasename := make([]byte, int(databaselUint64))
-	copy(databasename, content[8:8+int(databaselUint64)])
-
-	token := make([]byte, len(content)-8-int(databaselUint64))
-	copy(token, content[8+int(databaselUint64):])
+	dbnamelen := make([]byte, 8)
+	copy(dbnamelen, content[:8])
+	dbnamelenUint64 := binary.BigEndian.Uint64(dbnamelen)
+	if int(dbnamelenUint64) > len(content) {
```
Comment (jeroiraz) [inline pkg/client/token_service.go:99]: yes, actually should not be greater than remaining content, 8 bytes were already consumed when reading the encoded length

Comment (ellisab) [reply pkg/client/token_service.go:99]: @jeroiraz https://play.golang.org/p/WT-Nin6UVBb
checking that dbnamelenUint64. is not bigger than the whole token avoids the panic

I do not think that casting to int is needed  if uint64(len(content)) is used https://play.golang.org/p/cdH3lMq5TC0

Comment (jeroiraz) [reply pkg/client/token_service.go:99]: I mean something like
lenb := make([]byte, 8)
copy(lenb, content)

dbNameLen := int(binary.BigEndian.Uint64(lenb))

 `if dbNameLen > len(content)-8` ...  // note: here dbNameLen should be less than whole content less 8 bytes used to encode the len of the dbName itself

**Comment by jeroiraz** [comment]
Thanks a lot @ellisab for your contribution :)

**Pull request merged**